src/features/dashboard/components/ApiTokenTable/DeleteTokenDialog/index.tsx

```diff
@@ -1,57 +1,60 @@
-import React, { useMemo, useCallback } from 'react';
-import { Modal } from '@deriv/ui';
-import { TModalActionButton } from '@deriv/ui/dist/types/src/components/core/modal/types';
+import React, { useCallback, useContext, useEffect } from 'react';
+import { TTokenType } from '@site/src/types';
+import { Modal } from '@deriv-com/quill-ui';
+import useDeleteToken from '../../../hooks/useDeleteToken';
+import useDeviceType from '@site/src/hooks/useDeviceType';
+import { ApiTokenContext } from '@site/src/contexts/api-token/api-token.context';
+import './delete-token-dialog.scss';
 
-type TDeleteTokendialog = {
-  setToggleModal: React.Dispatch<React.SetStateAction<boolean>>;
-  onDelete: () => void;
+type TDeleteTokenDialogProps = {
+  token: TTokenType;
+  onClose: () => void;
 };
 
-const DeleteTokenDialog = ({ onDelete, setToggleModal }: TDeleteTokendialog) => {
-  const onOpenChange = useCallback(
-    (open: boolean) => {
-      if (!open) {
-        setToggleModal(false);
-      }
-    },
-    [setToggleModal],
-  );
+const DeleteTokenDialog = ({ token, onClose }: TDeleteTokenDialogProps) => {
+  const { deleteToken } = useDeleteToken();
+  const { deviceType } = useDeviceType();
+  const { tokens, updateTokens } = useContext(ApiTokenContext);
 
-  const actionButtons: TModalActionButton[] = useMemo(
-    () => [
-      {
-        id: 0,
-        text: 'Cancel',
-        color: 'secondary',
-        onClick: () => {
-          setToggleModal(false);
-        },
-      },
-      {
-        id: 1,
-        text: 'Yes, delete',
-        color: 'primary',
-        onClick: () => {
-          onDelete();
-        },
-      },
-    ],
-    [setToggleModal],
-  );
+  useEffect(() => {
+    document.body.style.overflow = 'hidden';
+    return () => {
+      document.body.style.overflow = '';
+    };
+  }, []);
+
+  const handleDelete = useCallback(() => {
+    deleteToken(token.token);
+    const newTokens = [];
+    for (let i = 0; i < tokens.length; i++) {
+      if (tokens[i].token !== token.token) {
+        newTokens.push(tokens[i]);
+      }
+    }
+    updateTokens(newTokens);
```

Comment:
Do we need to manually handle this from our side? If we make the delete API call shouldn't we get a response with updated token list? If so can we just update the list with that response?

Reply:
Fixed this as well as the updateTokens is part of the response from the API where it's updating the tokens after deletion, this didn't need to be handled manually.